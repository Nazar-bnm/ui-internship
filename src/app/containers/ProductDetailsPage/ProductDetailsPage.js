import React, {Component} from 'react';

class ProductDetailsPage extends Component {
  constructor(props) {
    super(props);
    const { match: { params } } = this.props;


    this.state = {
      product: {
        name: params.name,
        id: params.id
      }
    }
  }

  returnBack = () => {
    this.props.history.goBack();
  }
  
  render() {
    console.log('params', this.props.match);
    return(
      <div>
        <div>{this.state.product.name}</div>
        <div>{this.state.product.id}</div>
        <button onClick={() => this.returnBack()}>Go Back</button>
      </div>
    );
  }
}

export default ProductDetailsPage