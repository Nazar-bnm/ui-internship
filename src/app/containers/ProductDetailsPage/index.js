import React, { Component } from 'react';
import PropTypes from 'prop-types';

class ProductDetailsPage extends Component {
  constructor(props) {
    super(props);

    this.state = {
      product: {
        name: this.props.name,
        id: this.props.id,
      },
    };
  }

  returnBack() {
    return this.props.history.goBack();
  }

  render() {
    return (
      <div>
        <div>{this.state.product.name}</div>
        <div>{this.state.product.id}</div>
        <button onClick={() => this.returnBack()}>Go Back</button>
      </div>
    );
  }
}

ProductDetailsPage.propTypes = {
  history: PropTypes.object,
  name: PropTypes.string,
  id: PropTypes.string,
};

export default ProductDetailsPage;
