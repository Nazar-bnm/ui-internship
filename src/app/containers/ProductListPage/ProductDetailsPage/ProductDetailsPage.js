import React, {Component} from 'react';
import { Link } from 'react-router-dom';

class ProductDetailsPage extends Component {
  renderProductList(){
    this.products = [...this.props.products];
    return this.products.map(({ product, id }) => {
      return <li key={id}>{product}</li>
    });
  }
  
  render() {    
    console.log(this.props);
    const { products } = this.props;
    return products ? <ul>{this.renderProductList()}</ul> : null;
  }
}

export default ProductDetailsPage