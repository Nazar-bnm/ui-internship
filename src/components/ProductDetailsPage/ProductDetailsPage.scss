@import '../../scss/global_styles';

.go-back-btn {
  -webkit-appearance: none;
  border: none;
  outline: none;
  cursor: pointer;
}

.product-details {
  display: flex;
  justify-content: space-between;
  padding: 20px 0;
  

  &-images {
    flex-wrap: nowrap;

    .carousel-image {
      margin-right: 6px;
      max-width: 160px;
      height: 700px;
    }
  }

  &-description {
    margin-left: 1rem;
    width: 40%;

    .button {
      -webkit-appearance: none;
      text-align: center;
      cursor: pointer;
      color: $white;
      background-color: $black;
      font-family: inherit;
      font-size: 0.8rem;
      text-transform: uppercase;
  
      &:hover {
        color: $black;
        background-color: $white;
      }
    }

    &-accordion {
      margin-top: 2rem;
      height: 300px;

      .accordion-item__description {
        color: $dove-gray;
        font-size: 0.9rem;
        font-family: font(primary-light);
      }

      .accordion-item__header,
      .contact-block {
        height: 50px;
      }

      .contact-block {
        border-top: 1px solid $silver-chalice;
    
        &__title {
          color: $black;
        }
    
        &__item {
          transition: background-color linear 0.5s;
          padding: 3px;
          background-color: $black;
          color: $white;
    
          &:hover,
          &:active {
            background-color: $silver-chalice;
          }
    
          a {
            text-decoration: none;
          }
        }
    
        i.icon {
          font-size: 1rem;
          color: $white;
        }
    
        &__icons {
          align-items: center;
        }
      }
    }

    &__product-order {
      .product-order__heading {
        font-size: 2rem;
        line-height: 2rem;
        letter-spacing: 1px;
        padding-bottom: 1rem;
      }

      .product-order__description {
        font-size: 0.9rem;
      }

      .button {
        width: 180px;
        height: 40px;
      }
    }
  }

  .product-image__container {
    width: 60%;
  }

  .product-image__big-image-container {
    width: 100%;

    .product-image {
      width: 100%;
      height: auto;
    }
  }
  .product-image__small-image {
    height: 100%;
  }
}

@include media-for(tablet) {
  .product-details {
    .product-order__heading,
    .product-order__price {
      text-align: center;
    }

    &-images {
      width: 55%;
      flex-direction: column;

      .carousel-image {
        order: 2;
        max-width: unset;
        width: 100%;
        height: unset;

        .carousel-container {
          width: 100%;
          margin-top: 6px;

          .product-image {
            max-width: 120px;
            width: 100%;
            height: auto;
          }
        }
      }

      .product-image {
        width: 100%;
        height: auto;
      }
    }

    .product-image__container {
      width: 50%;
      height: unset;
    }

    .product-image__big-image-container {
      width: 100%;
      height: unset;
    }
    .button {
      width: 110px;
    }
  }
}

@include media-for(mobile-landscape) {
  .product-details {
    flex-direction: column;

    .product-image__container {
      width: 100%;
    }

    .product-details-description {
      max-width: unset;
      width: 90%;
      margin: 0 auto;
    }
  }
}
