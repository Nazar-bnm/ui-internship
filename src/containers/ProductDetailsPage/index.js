import React, { Component } from 'react';
import PropTypes from 'prop-types';

class ProductDetailsPage extends Component {
  constructor(props) {
    super(props);

    const { match: { params } } = this.props;

    this.state = {
      product: {
        name: params.name,
        id: params.id,
      },
    };
  }

  returnBack() {
    return this.props.history.goBack();
  }

  render() {
    return (
      <div>
        <div>{this.state.product.name}</div>
        <div>{this.state.product.id}</div>
        <button onClick={() => this.returnBack()}>Go Back</button>
      </div>
    );
  }
}

ProductDetailsPage.propTypes = {
  // Routing history for returning back
  history: PropTypes.object,
  // Route path info
  match: PropTypes.object,
  // receiwed odject name
  name: PropTypes.string,
  // receiwed odject id
  id: PropTypes.string,
};

export default ProductDetailsPage;
 