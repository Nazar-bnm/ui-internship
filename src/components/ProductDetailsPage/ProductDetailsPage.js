import React, { useState } from 'react';

import ProductImage from '../ProductImage';
import ProductOrder from '../ProductOrder';
import Accordion from '../Accordion';
import ContactDetails from '../Footer/ContactDetails';
import { accordionItemsData } from '../../config/dataItemsAccordion';
import Button from '../shared/Button';

import './ProductDetailsPage.scss';

export const CN = 'product-details';

const ProductDetailsPage = (props) => {
  const [isVerticalCarousel, setVerticalCarousel] = useState(window.innerWidth > 768);
  const { match: { params: { id } }, history: { goBack }, products } = props;
  const product = products.find(({ _id }) => _id === id);
  window.onresize = () => setVerticalCarousel(window.innerWidth > 768);
  if (products.length) {
    const {
      images, title, price, description
    } = product;
    const productImageData = images.map(({ claudinaryId }) => ({ src: `${process.env.IMAGE_URL}/${claudinaryId}` }));


    return (
      <>
        <Button
          className="go-back-btn"
          icon="arrow left"
          onClick={goBack}
        >
      Go Back
        </Button>
        <div className={`${CN} content`}>
          <ProductImage className={`${CN}-images`} images={productImageData} verticalCarousel={isVerticalCarousel} />
          <div className={`${CN}-description`}>
            <ProductOrder className={`${CN}-description__product-order`} title={title} price={`$${price}`} description={description} />
            <div className={`${CN}-description-accordion`}>
              <Accordion data={accordionItemsData} heightItem="100px" className={`${CN}__accordion`} />
              <ContactDetails title="share" />
            </div>
          </div>
        </div>
      </>
    );
  }

  return <> </>;
};

export default ProductDetailsPage;
