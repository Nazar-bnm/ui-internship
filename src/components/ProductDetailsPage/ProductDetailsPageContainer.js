import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { addToWishlist } from '../../actions/actionsWishlist';
import showMessage from '../../actions/actionsNotifications';


import ProductDetailsPage from './ProductDetailsPage';

const mapStateToProps = ({ productsReducer }) => ({
  products: productsReducer.products
});

const mapDispatchToProps = (dispatch) => ({
  addToWishlist: bindActionCreators(addToWishlist, dispatch),
  showMessage: bindActionCreators(showMessage, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(ProductDetailsPage);
