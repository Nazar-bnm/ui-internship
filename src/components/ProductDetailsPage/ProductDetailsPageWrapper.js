import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Button from '../shared/Button';
import ProductDetailsPage from './ProductDetailsPage';

class ProductDetailsPageWrapper extends Component {
  constructor(props) {
    super(props);
    const { match: { params } } = this.props;

    this.state = {
      product: {
        name: params.name,
        id: params.id
      }
    };
  }

  renderGoBackBtn = () => {
    const { history } = this.props;

    return (
      <Button
        customClass="go-back-btn"
        icon="arrow left"
        onClickFunction={history.goBack}
      >
      Go Back
      </Button>
    );
  }

  render() {
    const { product } = this.state;
    const productName = product.name;
    // const productId = product.id;
    const productId = product.id;
    console.log(this.props);

    return (
      <div>
        {productName}
        {productId}
        {this.renderGoBackBtn()}
        <ProductDetailsPage id={productId} />
      </div>
    );
  }
}

ProductDetailsPageWrapper.propTypes = {
  history: PropTypes.object.isRequired,
  match: PropTypes.object,
  name: PropTypes.string,
  id: PropTypes.string
};

ProductDetailsPageWrapper.defaultProps = {
  match: {},
  name: '',
  id: ''
};

export default ProductDetailsPageWrapper;
