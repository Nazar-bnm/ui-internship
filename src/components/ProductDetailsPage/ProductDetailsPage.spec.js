import React from 'react';
import { shallow } from 'enzyme';

import ProductDetailsPage from './ProductDetailsPage';

xdescribe('<ProductDetailsPage />', () => {
  test('should match snapshot', () => {
    const wrapper = shallow(<ProductDetailsPage />);
    expect(wrapper).toMatchSnapshot();
  });
});
