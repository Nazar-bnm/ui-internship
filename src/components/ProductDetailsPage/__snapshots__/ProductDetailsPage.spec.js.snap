// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductDetailsPage /> should match snapshot 1`] = `
<div
  className="product-details content"
>
  <ProductImage
    images={
      Array [
        Object {
          "src": "src/assets/img/content/new4.png",
        },
        Object {
          "src": "src/assets/img/content/new2.png",
        },
        Object {
          "src": "src/assets/img/content/new3.png",
        },
      ]
    }
  />
  <div
    className="product-details-right-aside"
  >
    <ProductOrder
      colors={
        Array [
          Object {
            "label": "color:",
            "value": "color:",
          },
          Object {
            "label": "white",
            "value": "white",
          },
          Object {
            "label": "pink",
            "value": "pink",
          },
          Object {
            "label": "black",
            "value": "black",
          },
        ]
      }
      description=""
      price=""
      quantity={
        Array [
          Object {
            "label": "qty:",
            "value": "qty:",
          },
          Object {
            "label": "1",
            "value": "1",
          },
          Object {
            "label": "2",
            "value": "2",
          },
          Object {
            "label": "3",
            "value": "3",
          },
          Object {
            "label": "4",
            "value": "4",
          },
          Object {
            "label": "5",
            "value": "5",
          },
          Object {
            "label": "6",
            "value": "6",
          },
          Object {
            "label": "7",
            "value": "7",
          },
          Object {
            "label": "8",
            "value": "8",
          },
          Object {
            "label": "9",
            "value": "9",
          },
          Object {
            "label": "10",
            "value": "10",
          },
        ]
      }
      sizes={
        Array [
          Object {
            "label": "size:",
            "value": "size:",
          },
          Object {
            "label": "s",
            "value": "s",
          },
          Object {
            "label": "m",
            "value": "m",
          },
          Object {
            "label": "l",
            "value": "l",
          },
          Object {
            "label": "xl",
            "value": "xl",
          },
        ]
      }
      title=""
    />
    <div
      className="product-details-right-aside-accordion"
    >
      <Accordion
        className="product-details__accordion"
        data={
          Array [
            Object {
              "description": " Architecto, earum temporibus quidem ex eaque unde harum sit, 
      deleniti tempore laboriosam neque cumque!Architecto, earum temporibus 
      quidem ex eaque unde harum sit, deleniti tempore laboriosam neque cumque!",
              "id": 1,
              "title": "description",
            },
            Object {
              "description": "Lorem ipsum dolor sit amet consectetur adipisicing elit. 
      ",
              "id": 2,
              "title": "additional info",
            },
            Object {
              "description": "Earum temporibus quidem ex eaque unde harum sit, 
      ",
              "id": 3,
              "title": "tags",
            },
          ]
        }
        heightItem="100px"
        labels={Array []}
        open={false}
        scroll={false}
        showAll={false}
      />
      <ContactDetails
        title="share"
      />
    </div>
  </div>
</div>
`;
